types/noteflight-client/index.d.ts

```diff
@@ -0,0 +1,664 @@
+         */
+        scale?: number;
+        /**
+         * This optional parameter may be used to control how playback is handled in the embed, and also whether instrument sounds are downloaded or not.
+         * There are three settings for this parameter. If this option is not provided, then the playback mode is determined by the default setting in the score.
+         */
+        playback?: 'normal' | 'audioTrack' | 'echo' | 'silent';
+        /**
+         * This optional parameter specifies the user's role with respect to the document.
+         * Only two settings are permitted. The default setting is reader.
+         */
+        role?: 'reader' | 'template';
+        /**
+         * This optional parameter specifies the mode in which the document is displayed.
+         */
+        displayMode?: 'strip' | 'paginated' | 'flow';
+        /**
+         * This parameter specifies the technology used to display the document.
+         * If omitted, the document is displayed with Flash if present, otherwise with HTML5.
+         */
+        app?: 'flash' | 'html5';
+        /**
+         * This parameter shows a watermark in the embed score.
+         * The URL of the watermark has to be passed in this parameter.
+         */
+        watermark?: string;
+        /**
+         * This parameter scales the watermark image.
+         */
+        watermarkScale?: number;
+        /**
+         * This parameter hides the playback controls.
+         * Default is false.
+         */
+        hidePlaybackControls?: boolean;
+        /**
+         * This parameter hides the full window button in the header of an embed.
+         * It also suppresses the display of links to the full score in the Sharing panel. Default is false.
+         */
+        hideFullWindow?: boolean;
+        /**
+         * This parameter, if set to false, hides the score's title, supertitle, subtitle, composer, and lyricist.
+         * Default is true.
+         */
+        showScoreInfo?: boolean;
+        /**
+         * Determines whether the window for synced media, if present, displays.
+         * Default is true.
+         */
+        showMediaTab?: boolean;
+    };
+}
+
+type EventType =
+    | 'any'
+    | 'editorReady'
+    | 'scoreDataLoaded'
+    | 'selectionChange'
+    | 'playbackRequest'
+    | 'playbackStop'
+    | 'noteSizeChange'
+    | 'pageSizeChange'
+    | 'partsTransposed';
+
+type NoteflightEvent<T> = NoteflightEventProps &
+    EventFilter<T, 'scoreDataLoaded', ScoreDataLoadedProps> &
+    EventFilter<T, 'selectionChange', SelectionChangeProps> &
+    EventFilter<T, 'playbackRequest' | 'playbackStop', PlaybackProps>;
+
+type EventFilter<T, Event, Props> = T extends Event ? Props : T extends 'any' ? Partial<Props> : {};
+
+interface NoteflightEventProps {
+    /**
+     * 	A string value that identifies the kind of event that was dispatched.
```

Comment:
<!--
  1 failure:  The root .editorc...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: types/noteflight-client/index.d.ts;
  Line: 127;
-->
- :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.